http/interceptor.go

```diff
@@ -40,33 +41,37 @@ func (i *rwInterceptor) WriteHeader(statusCode int) {
 		}
 	}
 
-	i.hasStatusCode = true
 	i.statusCode = statusCode
 	if it := i.tx.ProcessResponseHeaders(statusCode, i.proto); it != nil {
 		i.statusCode = obtainStatusCodeFromInterruptionOrDefault(it, i.statusCode)
 	}
+	i.hasStatusCode = true
 }
 
 func (i *rwInterceptor) Write(b []byte) (int, error) {
-	if !i.hasStatusCode {
-		i.WriteHeader(http.StatusOK)
-	}
 	if i.tx.IsInterrupted() {
 		// if there is an interruption it must be from phase 4 and hence
 		// we won't write anything to either the body or the buffer.
 		return 0, nil
 	}
 
-	if i.tx.IsResponseBodyAccessible() {
+	if !i.hasStatusCode {
+		i.WriteHeader(http.StatusOK)
```

Comment:
Didn't dig deep into the code, but because Go automatically sends OK if not explicitly sent, do we need the hasStatusCode logic?

Reply:
Removes that.